### DISCUSSION TITLE
Pagamento do Pix Retornando XXXX no Payer Email

### DISCUSSION THREAD

**Discussion by stefanobarcellos** [mercadopago/sdk-js#214] [Q&A] [answered]
Eu fiz todo o sistema me baseando no e-mail do cliente. Mas quando faço uma consulta em produção na id do pagamento de PIX aprovado, exemplo:

https://api.mercadopago.com/v1/payments/{$paymentId}

Recebo:

"order": {},
    "payer": {
        "email": "XXXXXXXXXXX",
        "entity_type": null,
        "first_name": null,
        "id": "<removido>",
        "identification": {
            "number": null,
            "type": null
        },
        "last_name": null,
        "operator_id": null,
        "phone": {
            "number": null,
            "extension": null,
            "area_code": null
        },
        "type": null
        
Eu gostaria que a API me retornasse o e-mail do cliente e não um XXXXXXXXXXX.

Estou usando as credenciais corretamente em produção, já verifiquei, mas ocorre isso. Verifiquei o código diversas vezes também.

Parece que compras pelo cartão ele retorna o e-mail normalmente.

Lá no painel do mercadopago eu consigo ver o e-mail de quem pagou com pix, mas a resposta da API é XXXXXXXXXXX.

Como posso corrigir isso? Alguma outra lógica? Existe alguma permissão que me esqueci para ele retornar o e-mail no pagamento pix?

Obrigado.

**Comment by icaldana** [comment, marked as answer]
Olá @stefanobarcellos!

Entende-se que o email do comprador nesse caso é uma informação sensível e por isso não é retornado. O que te recomendaria é atrelar em seu backend o `paymentId` com o email do pagador utilizado durante o envio da criação do pagamento.

Sobre esse valor `XXXXXXXXXXX` que a API está retornando, estou em conversa com o time responsável para melhorar essa experiência, agradeço pelo seu feedback.

Obrigado!

**Comment by stefanobarcellos** [comment]
Fiz o registro do e-mail pelo backend do servidor mesmo, aí não enfrentei mais esse problema de dados sensíveis.
